numpyro/mcmc.py

```diff
@@ -738,7 +738,7 @@ def _single_chain_mcmc(self, init, collect_fields=('z',), collect_warmup=False,
         if len(collect_fields) == 1:
             samples = (samples,)
         samples = dict(zip(collect_fields, samples))
-        samples['z'] = vmap(constrain_fn)(samples['z'])
+        samples['z'] = vmap(constrain_fn)(samples['z']) if len(tree_flatten(samples)[0]) > 0 else samples['z']
```

Comment:
if `samples['z']` is empty does vmap complain?

Reply:
Yes, `vmap` will complain about it: `import jax; jax.vmap(lambda x: x)({})` triggers
```
ValueError: not enough values to unpack (expected 1, got 0)
```